### PR TITLE
feat(cli): intro warning (36) support for comments

### PR DESCRIPTION
**Pull request by Druue**
closes https://github.com/prisma/schema-team/issues/388

### CODE REVIEW

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 1]
```diff
@@ -483,7 +483,7 @@ describe('postgresql - missing database', () => {
   })
 })
 
-describeIf(process.platform != 'win32', 'postgresql views fs I/O', () => {
+describeIf(process.platform != 'win32')('postgresql views fs I/O', () => {
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:486]: was that a bug?

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 2]
```diff
@@ -1582,11 +1582,11 @@ describe('postgresql introspection stopgaps', () => {
     })
   }
 
-  describe('postgres partitioned', () => {
+  describe('warning 27', () => {
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:1585]: I think worthwhile to keep the "topic" if the warning code here as well.
```suggestion
  describe('warning 27 - postgres partition tables', () => {
```

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 3]
```diff
@@ -2604,12 +2755,59 @@ describeIf(!process.env.TEST_SKIP_COCKROACHDB)('cockroachdb stopgaps', () => {
...
+                                                                                                      `)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+  })
+
+  describe('warning 36/5 - comments found: models & fields', () => {
+    const stopGap = 'comments'
+    const warningCode = 36
+    const variant = 5
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:2772]: What does the variant actually identify here? 
It's weird that the variants are spread across different databases, isn't it?
Isn't this variant 1 for Cockroach?

File: packages/migrate/src/__tests__/DbPull.test.ts (hunk 1 above)
Comment (Druue) [reply packages/migrate/src/__tests__/DbPull.test.ts:486]: Yeah

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 4]
```diff
@@ -1850,6 +1850,156 @@ describe('postgresql introspection stopgaps', () => {
...
+          // - Type "view", Name "b"
+          // - Type "field", Name "b.val"
+          // 
+        `)
+        expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+        expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      })
+    })
+
+    describe('36/3 - comments found: types', () => {
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:1959]: enums only, right? dont think we support other types
```suggestion
    describe('36/3 - comments found: enums', () => {
```

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 5]
```diff
@@ -2677,6 +2875,96 @@ describe('mysql', () => {
   })
 })
 
+describe('mysql introspection stopgaps', () => {
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:2878]: Shouldn't this be inside the `describe(mysql, ...)` above?

File: packages/migrate/src/__tests__/DbPull.test.ts (hunk 4 above)
Comment (Druue) [reply packages/migrate/src/__tests__/DbPull.test.ts:1959]: Yep, this is just enums (in engines) for now, can change this to be less generic if that's what we want

File: packages/migrate/src/__tests__/DbPull.test.ts (hunk 1 above)
Comment (janpio) [reply packages/migrate/src/__tests__/DbPull.test.ts:486]: yay then! also explains the unrelated snapshots being added.

File: packages/migrate/src/__tests__/DbPull.test.ts (hunk 3 above)
Comment (janpio) [reply packages/migrate/src/__tests__/DbPull.test.ts:2772]: confused me a tiny bit, but doesn't matter much as long as the test is there.

**Review by janpio — APPROVED** [review]

File: packages/migrate/src/__tests__/fixtures/introspection/cockroachdb/comments-warning-36-5/setup.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE a (
+    id INT PRIMARY KEY,
+    val VARCHAR(20)
+);
+
+COMMENT ON TABLE a IS 'vof';
+COMMENT ON COLUMN a.val IS 'mjau';
```
Comment (jkomyno) [inline packages/migrate/src/__tests__/fixtures/introspection/cockroachdb/comments-warning-36-5/setup.sql:7]: (I lol'ed at this)

File: packages/migrate/src/types.ts [hunk 7]
```diff
@@ -253,6 +254,11 @@ export namespace EngineArgs {
     columnName: string
   }
 
+  type AffectedObject = {
+    type: string
+    name: string
+  }
```
Comment (jkomyno) [inline packages/migrate/src/types.ts:260]: A comment at the top of this object type declaration would be nice

**Review by jkomyno — APPROVED** [review]
Approved with a nit

File: packages/migrate/src/types.ts (hunk 7 above)
Comment (Druue) [reply packages/migrate/src/types.ts:260]: What kind of comment would you expect here that you would find helpful?

Comment (Druue) [reply packages/migrate/src/types.ts]: Updated to:
```ts
// This represents any database object
// Model, enum, view, column, etc.
type AffectedObject = {
  type: string // Tells us what kind of database object
  name: string
}
```

**Pull request merged**